bdk_core/src/sparse_chain.rs

```diff
@@ -310,13 +309,16 @@ impl<I: ChainIndex> SparseChain<I> {
             .map(|(_, txid)| (*txid, None))
             .collect();
 
-        let changeset = ChangeSet::<I> {
+        ChangeSet::<I> {
             txids,
             ..Default::default()
-        };
+        }
+    }
 
-        self.apply_changeset(&changeset);
-        changeset
+    /// Clear the mempool list. Use with caution.
+    pub fn clear_mempool(&mut self) {
```

Comment:
If my understanding regarding persistence is correct, should we also return the changeset here as well?

Reply:
https://github.com/LLFourn/bdk_core_staging/pull/69#discussion_r1032748972